sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/serialization/CoderBasedKafkaSerializer.java

```diff
@@ -20,29 +20,79 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.common.collect.ImmutableMap;
+
+import java.io.InputStream;
 import java.util.Map;
 import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.coders.AvroCoder;
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.io.kafka.KafkaIO;
 import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.kafka.common.serialization.Deserializer;
 import org.apache.kafka.common.serialization.Serializer;
 
 /**
- * Implements Kafka's {@link Serializer} with a {@link Coder}.
+ * Implements a Kafka {@link Deserializer} and {@link Serializer} with a {@link Coder}.
```

Comment:
I'd like to remove this in the current PR and before FSR since, as @aviemzur points out, this is against the style guide. Especially since, as you point out, there exists a Kafka (de)serializer for Avro. Then the only case that could present any problem is protobufs - but I suppose, if the Kafka community still hasn't implemented support for them, it means they're not that widely used in the Kafka community.